.github/workflows/github-page-publish.yml

```
@@ -0,0 +1,23 @@
+name: Github Page Publish
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

I think we only need to deploy the reference documentation when a new commit is pushed into the main branch.

```suggestion
  push:
    branches: [ main ]
```